js/build.js

```diff
@@ -372,22 +372,32 @@ $('[data-login-id]').each(function() {
     return Promise.all(promises);
   }
 
+  function getUserFromSession(session) {
+    if (!session.entries || !session.entries.flipletLogin) {
+      throw new Error('Login session not found');
+    }
+
+    return session.entries.flipletLogin;
+  }
+
   function init() {
     _this.loginPV = {};
     Fliplet.User.getCachedSession()
       .then(function(session) {
-        if (!session || !session.user) {
+        if (!session || !session.server || !session.server.flipletLogin) {
           return Promise.reject('Login session not found');
         }
 
+        var loggedUser = getUserFromSession(session);
+
         // Update stored email address based on retrieved session
         return updateUserData({
-          id: session.user.id,
+          id: loggedUser.id || session.user.id,
           region: session.auth_token.substr(0, 2),
-          userRoleId: session.user.userRoleId,
+          userRoleId: loggedUser.userRoleId || session.user.userRoleId,
           authToken: session.auth_token,
-          email: session.user.email,
-          legacy: session.legacy
+          email: loggedUser.email || session.user.email,
+          legacy: loggedUser.legacy || session.legacy
```

Comment:
@squallstar Can you verify if the legacy property is non-empty when logging in with an account with a legacy v1 user? If it's empty then Fliplet Viewer will crash when opening v1 apps.

Reply:
@tonytlwu will do, is this for FV 2.0? If so, my guess is I need to rename `loggedUser.legacy` to `legacyId` and make it a boolean

Reply:
@squallstar `legacy` is expected to be an object

https://github.com/Fliplet/fliplet-ios-version-2/blob/f3b74606b74672206a91e19dadec8c79685b8b74/Fliplet/Classes/FLTAuthenticationManager.m#L50-L54

Reply:
It might require a server side change because that could be why I was using `session.user`. It's all coming back to me now 🥇

Reply:
ahhh thanks for the heads up, @tonytlwu yeah I think it will need a server-side change. I'll look into this.